### PR TITLE
implement imgur.io resolving

### PR DESCRIPTION
**Pull request by gempir**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Allow imgur.io to urls to be resolved as per #361

Most of the work done by @leon-richardt of sending a PR to our go imgur client.

### CODE REVIEW

File: internal/resolvers/imgur/resolver.go [hunk 1]
```diff
@@ -27,7 +29,19 @@ type Resolver struct {
...
+			return nil, result
+		}
+
+		result = utils.IsDomain(url, domain)
+		if result {
+			return nil, result
+		}
+	}
+
+	return ctx, false
```
Comment (leon-richardt) [inline internal/resolvers/imgur/resolver.go:44]: This seems to be causing the error you're running into. The `Context` returned by the method is later used in `internal/resolvers/default/link_resolver.go` and will cause an error if it's `nil`. The error can be resolved with this change:

```suggestion
	for _, domain := range VALID_IMGUR_DOMAINS {
		result := utils.IsSubdomainOf(url, domain)
		if result {
			return ctx, result
		}

		result = utils.IsDomain(url, domain)
		if result {
			return ctx, result
		}
	}

	return ctx, false
```

---

However, even after making this change, imgur.io domains will not be resolved correctly. [It seems that `koffeinsource/go-imgur` does not support that domain yet](https://github.com/koffeinsource/go-imgur/blob/master/fromURL.go#L20-L44).

Comment (leon-richardt) [reply internal/resolvers/imgur/resolver.go:44]: Small update: I have created https://github.com/koffeinsource/go-imgur/pull/16 at the upstream repository to raise awareness.

**Comment by leon-richardt** [comment]
https://github.com/koffeinsource/go-imgur/pull/16 has been merged upstream and is included in the `v0.4.1` release.
@gempir, if you rebase your branch on `master` and bump the version in `go.mod`, it should Just Work:tm:.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,7 @@
 - Dev: Improve FrankerFaceZ tests. (#295)
 - Dev: Improve Livestreamfails tests. (#297, #301)
 - Dev: Improve default resolver tests. (#300)
+- Dev: Resolve imgur.io links (#365)
```
Comment (pajlada) [inline CHANGELOG.md:28]:
```suggestion
- Dev: Resolve imgur.io links. (#365)
```

**Review by pajlada — CHANGES_REQUESTED** [review]

**Review by pajlada — APPROVED** [review]

**Pull request merged**